forch/faucetizer.py

```diff
@@ -35,14 +66,12 @@ def process_network_state(self, network_state):
         self._faucetize(macs)
 
     def _process_device_learning(self, mac, learning):
-        device_map = self._devices.setdefault(mac, {})
-        device_map['new_switch'] = learning.switch
-        device_map['new_port'] = learning.port
+        device = self._devices.setdefault(mac, Device())
+        device.new_learning = learning
 
     def _process_device_behavior(self, mac, behavior):
-        device_map = self._devices.setdefault(mac, {})
-        device_map['vlan'] = behavior.vid
-        device_map['role'] = behavior.role
+        device = self._devices.setdefault(mac, Device)
+        device.behavior = behavior
```

Comment:
how is this a direct attribute modifier?  I don't see a "behavior" attribute on the class?

Reply:
Changed to explicit attributes